tools/wpt/browser.py

```diff
@@ -516,264 +516,201 @@ def version(self, binary=None, webdriver_binary=None):
+        if self.platform is None:
+            raise ValueError(
+                "Unable to construct a valid Chromium package name for current platform")
 
-        if (platform == "Linux" or platform == "Win") and uname[4] == "x86_64":
-            platform += "_x64"
+        if (self.platform == "Linux" or self.platform == "Win") and uname[4] == "x86_64":
+            return f"{self.platform}_x64"
 
-        return platform
+        return self.platform
 
     def _chromium_package_name(self):
-        return "chrome-%s" % self.platforms.get(uname[0]).lower()
+        return f"chrome-{self.platform.lower()}"
 
-    def _latest_chromium_snapshot_url(self):
-        # Make sure we use the same revision in an invocation.
+    def _format_chromium_snapshot_url(self, revision=None):
         architecture = self._chromium_platform_string()
-        if self._last_change is None:
-            revision_url = "https://storage.googleapis.com/chromium-browser-snapshots/%s/LAST_CHANGE" % architecture
-            self._last_change = get(revision_url).text.strip()
-        return "https://storage.googleapis.com/chromium-browser-snapshots/%s/%s/" % (architecture, self._last_change)
+        # Make sure we use the same revision in an invocation.
+        # If we have a last revision used, that revision takes priority.
+        if self.last_revision_used is None and revision is None:
+            # No revision provided or previously used. use latest revision.
+            self.revision_attempted = self._get_latest_chromium_revision(architecture)
+        elif revision is not None:
+            self.revision_attempted = revision
+        else:
+            self.revision_attempted = self.last_revision_used
 
-    def find_nightly_binary(self, dest):
-        if uname[0] == "Darwin":
-            return find_executable("Chromium",
-                                   os.path.join(dest, self._chromium_package_name(), "Chromium.app", "Contents", "MacOS"))
-        # find_executable will add .exe on Windows automatically.
-        return find_executable("chrome", os.path.join(dest, self._chromium_package_name()))
+        # The attempted revision is kept. If it is successfully used,
+        # it will be used again if another component is also installed (browser/webdriver/mojojs).
+        return ("https://storage.googleapis.com/chromium-browser-snapshots/"
+                f"{architecture}/{self.revision_attempted}/")
 
-    def find_binary(self, venv_path=None, channel=None):
-        if channel == "nightly":
-            return self.find_nightly_binary(self._get_dest(venv_path, channel))
+    def _get_chromium_url_by_version(self, version=None):
+        if version and version.lower() != "latest":
+            try:
+                detected_revision = self._get_revision_from_version(version)
+                url = f"{self._format_chromium_snapshot_url(detected_revision)}"
```

Comment:
This is equivalent, right?
```suggestion
                url = self._format_chromium_snapshot_url(detected_revision)
```

Reply:
Yup, this is a remnant from an old implementation adding the file name to the string - cleaned it up!